File: Builder/button/PrimaryButton.java
package button;

public class PrimaryButton extends Button {

    private PrimaryButton(ButtonBuilder builder) {
        super(builder);
    }

    @Override
    public void render() {
        for (int i = 0; i < width; i++)
            System.out.print("-");

        for (int i = 0; i < height; i++) {
            if (i == height / 2)
                System.out.println(text);
            else
                System.out.println();
        }

        for (int i = 0; i < width; i++)
            System.out.print("-");
        System.out.println();
    }

    @Override
    public void onClick() {
        System.out.println("PrimaryButton: " + action);
    }

    public static class Builder extends ButtonBuilder {
        public Builder(String action) {
            this.action = action;
        }

        @Override
        public Button build() {
            return new PrimaryButton(this);
        }
    }
}

File: BuilderKotlin/button/PrimaryButton.kt
package button

class PrimaryButton(private val builder: ButtonBuilder) : Button(builder) {
    override fun render() {
        for (i in 0 until width!!) print("-")

        for (i in 0 until height!!) {
            if (i == height!! / 2) println(text) else println()
        }

        for (i in 0 until width!!) print("-")
        println()
    }

    override fun onClick() {
        println("PrimaryButton: $action")
    }

    class Builder(action: String) : ButtonBuilder(action = action) {
        override fun build() = PrimaryButton(this)
    }
}